tests/hooks/useSynchronizedRequest.test.tsx

```diff
@@ -0,0 +1,26 @@
+import { useSynchronizedRequest } from '../../src/hooks/useSynchronizedRequest';
+import { renderHook } from '@testing-library/react-hooks';
+import { waitFor } from '@testing-library/react';
+
+
+let requestFunction = async () => 0;
+const errorFunction = () => 'error';
+
+it('returns an updated function with the same reference', async () => {
+  const { result, rerender } = renderHook(() => useSynchronizedRequest(requestFunction, errorFunction));
```

Comment:
in this particular test I would probably omit errorFunction since it's not being used by anything

Reply:
removed